### PR TITLE
feat: change build system to tsup

### PR DESCRIPTION
**Pull request by favna**
Finally one and the same build system for all packages

### CODE REVIEW

File: packages/async-queue/package.json [hunk 1]
```diff
@@ -19,8 +19,7 @@
 	"scripts": {
 		"test": "jest",
 		"lint": "eslint src tests --ext ts --fix -c ../../.eslintrc",
-		"build": "rollup -c",
-		"start": "yarn build -w",
+		"build": "tsup && tsc -b src",
```
Comment (vladfrangu) [inline packages/async-queue/package.json:22]: Doesn't this override...the tsup index.js?

File: packages/fetch/tsup.config.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import { createTsupConfig } from '../../scripts/tsup.config';
+
+export default createTsupConfig({ globalName: 'SapphireFetch' });
```
Comment (vladfrangu) [inline packages/fetch/tsup.config.ts:3]: Are we sure tsup deals with cross-fetch correctly for UMD?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/async-queue/package.json (hunk 1 above)
Comment (vladfrangu) [reply packages/async-queue/package.json:22]: Wait, emitDeclarationOnly. Woops

File: packages/fetch/tsup.config.ts (hunk 2 above)
Comment (favna) [reply packages/fetch/tsup.config.ts:3]: Why do you think I labeled this "needs testing" LMAO.

File: packages/async-queue/package.json (hunk 1 above)
Comment (favna) [reply packages/async-queue/package.json:22]: No because it's set to `emitDeclarationOnly`

File: packages/fetch/tsup.config.ts (hunk 2 above)
Comment (favna) [reply packages/fetch/tsup.config.ts:3]: Okay it definitely does not. It goes:
```js
const result = await (0, import_cross_fetch.fetch)(stringUrl, options);
```
`import_cross_fetch.fetch` is
```js
var import_cross_fetch = __toESM(require_node_ponyfill());
```
```js
var require_node_ponyfill = __commonJS({
    "../../node_modules/cross-fetch/dist/node-ponyfill.js"(exports, module) {
      var nodeFetch2 = (init_lib(), __toCommonJS(lib_exports));
      var realFetch = nodeFetch2.default || nodeFetch2;
```

So I'll have to come up with a plan for this.

**Review by vladfrangu — DISMISSED** [review]

**Comment by favna** [comment]
@sapphiredev pack

File: packages/fetch/tsup.config.ts (hunk 2 above)
Comment (favna) [reply packages/fetch/tsup.config.ts]: Tested latest commit with
- `index.mjs` (ESM)
- `index.cjs` (CJS)
- iife (through `index.html` with `<script src="./dist/index.global.js></script>` in `head` and a fetch in body)
- webpack (through create-react-app)
- swc (through NextJS)

All finally working. It took reverting back to cross-fetch and rollup which is a big shame but I have no idea what else to do. No matter what I tried, one thing or the other just didn't work. I have no idea how we're ever going to do undici for this package.

**Comment by favna** [comment]
@sapphiredev pack

**Review by favna — COMMENTED** [review]
_Tested fetch_

Webpack: ✅
Swc: ✅
ESM: ✅
CJS: ✅
iife: ✅

_Tested other packages_
(Using Dragonite)
```
"@sapphire/async-queue": "portal:/F:/dev/sapphiredev/utilities/packages/async-queue",
"@sapphire/decorators": "portal:/F:/dev/sapphiredev/utilities/packages/decorators",
"@sapphire/discord-utilities": "portal:/F:/dev/sapphiredev/utilities/packages/discord-utilities",
"@sapphire/discord.js-utilities": "portal:/F:/dev/sapphiredev/utilities/packages/discord.js-utilities",
"@sapphire/eslint-config": "portal:/F:/dev/sapphiredev/utilities/packages/eslint-config",
"@sapphire/fetch": "portal:/F:/dev/sapphiredev/utilities/packages/fetch",
"@sapphire/prettier-config": "portal:/F:/dev/sapphiredev/utilities/packages/prettier-config",
"@sapphire/ratelimits": "portal:/F:/dev/sapphiredev/utilities/packages/ratelimits",
"@sapphire/stopwatch": "portal:/F:/dev/sapphiredev/utilities/packages/stopwatch",
"@sapphire/time-utilities": "portal:/F:/dev/sapphiredev/utilities/packages/time-utilities",
"@sapphire/utilities": "portal:/F:/dev/sapphiredev/utilities/packages/utilities"
```

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**